python/src/nanoarrow/_lib.pyx

```diff
@@ -478,6 +593,130 @@ cdef class CSchemaView:
     def __repr__(self):
         return _lib_utils.schema_view_repr(self)
 
+
+cdef class CSchemaBuilder:
+    cdef CSchema c_schema
+    cdef ArrowSchema* _ptr
+
+    def __cinit__(self, CSchema schema):
+        self.c_schema = schema
+        self._ptr = schema._ptr
+        if self._ptr.release == NULL:
+            ArrowSchemaInit(self._ptr)
+
+    @staticmethod
+    def allocate():
+        return CSchemaBuilder(CSchema.allocate())
+
+    def child(self, int64_t i):
+        return CSchemaBuilder(self.c_schema.child(i))
+
+    def set_type(self, int type):
+        self.c_schema._assert_valid()
+
+        cdef int result = ArrowSchemaSetType(self._ptr, <ArrowType>type)
+        if result != NANOARROW_OK:
+            Error.raise_error("ArrowSchemaSetType()", result)
+
+        return self
+
+    def set_type_decimal(self, int type, int precision, int scale):
```

Comment:
I don't think it matters for cython but as a matter of style `type` is a reserved keyword in Python, so you typically don't use it as a variable name. I would suggest `type_` instead